design/improve_error_handling_todo_list.md

```diff
@@ -0,0 +1,218 @@
+# Improve Error Handling - TODO List
+
+
+This doucment maps the builtin or third party exceptino that are raised and suggest which exceptions we should use instead. It is listed by files, inside of file function and inside of function an error handling flow. In addition information from this document about reasons of why we throw these exception should be part of the relevant functions in their documentation.
+
+* api.py
+
+  * init function - These two exceptions can be replaced with MissingParameterException
+
+    * EmptyParameterError Account
+
+      ```python
+      self._account = account
+      if not self._account:
+          raise RuntimeError("Account cannot be empty!")
+      ```
+
+    * EmptyParameterError Url
+
+      * ```python
+        self._account = account
+        if not self._account:
+            raise RuntimeError("Account cannot be empty!")
+        ```
+
+  * login function-
+
+    * MissingRequiredParameterException
+
+      ```python
+      if not login_id or not password:
+          # TODO: Use custom error
+          raise RuntimeError("Missing parameters in login invocation!")
+      ```
+
+  * authenticate function
+
+    * MissingRequiredParameterException
+
+      * ```python
+        if not self.login_id or not self.api_key:
+            # TODO: Use custom error
+            raise RuntimeError("Missing parameters in authentication invocation!")
+        ```
+
+  * rotate_other_api_key function
+
+    * UnrotatableResource - We can rotate only user or host api key
+
+      ```python
+      if resource.type not in ('user', 'host'):
+          raise Exception("Error: Invalid resource type")
+      ```
+
+* client.py
+
+  * init function
+
+    * CredentialsParseError
```

Comment:
Exception is the standard I am pretty sure. Not `Error` at the end

Reply:
Changed to exception